### PR TITLE
DOC: Added examples for union_categoricals

### LINKED ISSUE #16390: DOC: union_categoricals could use some examples in the doc-string
can prob just use these: http://pandas.pydata.org/pandas-docs/stable/categorical.html#unioning

also adding a link in the doc-string to these docs would be nice

### PR DESCRIPTION
**Pull request by GuessWhoSamFoo**
- [x] closes #16390 
 - [ ] tests added / passed
 - [x] passes ``git diff upstream/master --name-only -- '*.py' | flake8 --diff``
 - [ ] whatsnew entry

Copied and posted over examples into the doc string; added link.

### CODE REVIEW

File: pandas/core/dtypes/concat.py [hunk 1]
```diff
@@ -242,6 +242,72 @@ def union_categoricals(to_union, sort_categories=False, ignore_order=False):
         - sort_categories=True and Categoricals are ordered
     ValueError
         Empty list of categoricals passed
+
+    Notes
+    -----
+
+    To learn more about categories, see 'link
```
Comment (jreback) [inline pandas/core/dtypes/concat.py]: see `here .....`

and the ``__`` (at the end) should be *outside* the single quote

File: pandas/core/dtypes/concat.py [hunk 2]
```diff
@@ -242,6 +242,72 @@ def union_categoricals(to_union, sort_categories=False, ignore_order=False):
...
+
+    By default, the resulting categories will be ordered as they appear
+    in the data. If you want the categories to be lexsorted, use
+    `sort_categories=True` argument.
+
+    >>> union_categoricals([a, b], sort_categories=True)
+    [b, c, a, b]
+    Categories (3, object): [a, b, c]
+
+    `union_categoricals` also works with the case of combining two
```
Comment (jreback) [inline pandas/core/dtypes/concat.py]: @TomAugspurger @jorisvandenbossche do we quote like this in a doc-string?

**Review by jreback — APPROVED** [review]
minor comments

File: pandas/core/dtypes/concat.py (hunk 1 above)
Comment (jorisvandenbossche) [inline pandas/core/dtypes/concat.py]: and the starting quote should also be backtick (`` ` ``), not a single quote (like you did for the ending quote)

File: pandas/core/dtypes/concat.py [hunk 3]
```diff
@@ -242,6 +242,72 @@ def union_categoricals(to_union, sort_categories=False, ignore_order=False):
...
+    categories being combined.
+
+    >>> a = pd.Categorical(["b", "c"])
+    >>> b = pd.Categorical(["a", "b"])
+    >>> union_categoricals([a, b])
+    [b, c, a, b]
+    Categories (3, object): [b, c, a]
+
+    By default, the resulting categories will be ordered as they appear
+    in the data. If you want the categories to be lexsorted, use
```
Comment (jorisvandenbossche) [inline pandas/core/dtypes/concat.py]: "resulting categories will be ordered as they appear in the data" is not fully correct (or at least subjective misinterpretable). 
The final categories are not ordered as they appear in the data, but as they appear in the `categories` of the data (ant this is not necessarily the same, as upon construction the categories are sorted, not in order of appearance).

Eg.

```
In [37]: a = pd.Categorical(["c", "b"])

In [38]:  b = pd.Categorical(["a", "b"])

In [39]:  pd.api.types.union_categoricals([a, b])
Out[39]: 
[c, b, a, b]
Categories (3, object): [b, c, a]
```

File: pandas/core/dtypes/concat.py [hunk 4]
```diff
@@ -242,6 +242,72 @@ def union_categoricals(to_union, sort_categories=False, ignore_order=False):
...
+
+    Examples
+    --------
+
+    If you want to combine categoricals that do not necessarily have
+    the same categories, `union_categoricals` will combine a list-like
+    of categoricals. The new categories will be the union of the
+    categories being combined.
+
+    >>> a = pd.Categorical(["b", "c"])
```
Comment (jorisvandenbossche) [inline pandas/core/dtypes/concat.py]: can you add a `from pandas.api.types import union_categorical`  at the beginning?

File: pandas/core/dtypes/concat.py (hunk 2 above)
Comment (jorisvandenbossche) [inline pandas/core/dtypes/concat.py]: I think this is OK (I don't think we consistently follow strict guidelines, but the numpydoc docstring explanation says to use single backticks to refer the keyword arguments or functions)

**Review by jorisvandenbossche — COMMENTED** [review]
Nice examples!

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
@GuessWhoSamFoo Thanks!

**Pull request merged**